package-res/reportviewer/reportviewer-prompt.js

```diff
@@ -35,6 +35,26 @@ define(['common-ui/util/util', 'pentaho/common/Messages', "dijit/registry", 'com
         return _api;
       },
 
+      _buildReportContentOptions: function(renderMode, defaultNameToAction) {
+        var options = util.getUrlParameters();
+
+        if (this.panel) {
+          $.extend(options, this.api.operation.getParameterValues());
+        }
+
+        options['renderMode'] = renderMode;
+
+        // SimpleReportingComponent expects name to be set
+        if (defaultNameToAction && options['name'] === undefined) {
+          options['name'] = options['action'];
```

Comment:
@krivera-pentaho probably not necessary creating _buildReportContentOptions function with additional boolean argument "defaultNameToAction" (better separate between 2 functions). For example, we can write the function without setting options['name'] because this function uses in one place in reportviewer-prompt.js.

```javascript
_buildReportContentOptions: function(renderMode) {
  var options = util.getUrlParameters();
  if (this.panel) {
    $.extend(options, this.api.operation.getParameterValues());
  }
  options['renderMode'] = renderMode;

  // Never send the session back. This is generated by the server.
  delete options['::session'];

  return options;
},
```

And additionally we can move the logic of setting options['name'] to reportviewer.js file to function "_buildReportContentOptions". For example:

```javascript
_buildReportContentOptions: function() {
  var options = this.reportPrompt._buildReportContentOptions('REPORT');
  if (options['name'] === undefined) {
    options['name'] = options['action'];
  }
  return options;
},
```

What do you think?